xclim/sdba/adjustment.py

```diff
@@ -135,6 +140,44 @@ def _make_dataset(self, **kwargs):
         self.ds = xr.Dataset(data_vars=kwargs)
         self.ds.attrs["adj_params"] = str(self)
 
+    def save_training(
+        self, filename: Optional[Union[Path, str]] = None, tempdir: Optional[str] = None
+    ):
+        """Save training data to a (temporary) file.
+
+        Save to a temporary file if `filename` is not given. The file will be
+        deleted when this Adjustment instance is deleted.
+
+        The dataset is immediately reload from file. This is meant to help divide dask's
+        workload when needed.
+
+        Parameters
+        ----------
+        filename : Optional[Union[Path, str]]
+          Filename of the saved file. When given, the file is not considered "temporary"
+          and is not deleted when the Adjustment object is deleted by Python.
+        tempdir : Optional[str]
+          The path to a directory where to save the temporary file. Ignored if `filename`
+          is given.
```

Comment:
I would simplify that to one single argument (e.g. path). What are the benefits of splitting the directory and the filename ?

Reply:
Simply for the option of making it a temporary file or not.
Example: on our server at ouranos it often makes more sense to save temporary files to `/exec/me/` than to `/tmp`.  But I still wanted the option to save to a persistent file for later inspection/re-use. And, maybe one day, a `Adjustment.from_training_ds('...')` class method.